tools/fuse_converter.py

```
@@ -1225,13 +1265,38 @@ def convert_input(input_path: Path, out_root: Path, kind: str, clean_output: boo
+                    "unsupported-already-native",
+                )
+            elif detected == "code":
+                report.add(
+                    "ERROR",
+                    "Compiled code mod detected. The converter only translates RailLoader JSON data and cannot reproduce DLL behavior; install a compatible code mod directly or ask its author for a FUSE-native version.",
+                    working_source,
+                    "unsupported-code-package",
+                )
             else:
-                raise RuntimeError("could not detect package type; use --kind route, --kind audio, or --kind asset")
+                raise RuntimeError("no convertible RailLoader JSON data detected; use the FUSE installer for code, asset, map-tile, or native packages")
```

Ruff (0.16.1)

[warning] Abstract `raise` to an inner function

(TRY301)